inst/compareScenarios/cs_01_stockFlows.Rmd

```diff
@@ -47,3 +47,103 @@ walk(heatingRes, showLinePlots, data = data)
+showAreaAndBarPlots(data, vars = subsec, tot = tot, orderVars = "user", scales = "fixed")
+```
+
+### All buildings
+
+### Residential
+```{r residential construction vars}
+tot <- "Construction|Residential"
+heatingRes <- paste(tot, heating, sep = "|")
+typeRes <- paste(tot, type, sep = "|")
+locationRes <- paste(tot, location, sep = "|")
+```
+
+#### by building type
+```{r Construction-Residential-(type)}
+showAreaAndBarPlots(data, vars = typeRes, tot = tot, orderVars = "user", scales = "fixed")
+walk(typeRes, showLinePlots, data = data)
+```
+
+#### by location
+```{r Construction-Residential-(location)}
+showAreaAndBarPlots(data, vars = locationRes, tot = tot, orderVars = "user", scales = "fixed")
+walk(locationRes, showLinePlots, data = data)
+```
+
+#### by Space heating
+```{r Construction-Residential-(heating)}
+showAreaAndBarPlots(data, vars = heatingRes, tot = tot, orderVars = "user", scales = "fixed")
+walk(heatingRes, showLinePlots, data = data)
+```
+
+### Commercial
+
+## Renovation flow
+
+```{r stock vars}
+tot <- "Renovation|Buildings"
+subsec <- c("Renovation|Residential",
+            "Renovation|Commercial")
+```
+
+```{r RenovationSubsector}
+showAreaAndBarPlots(data, vars = subsec, tot = tot, orderVars = "user", scales = "fixed")
+```
+
+### All buildings
+
+### Residential
+```{r residential Renovation vars}
+tot <- "Renovation|Residential"
+totBefore <- paste(tot, "Initial", sep = "|")
+totAfter <- paste(tot, "Final", sep = "|") # for proper plotting
+totAfterZero <- paste(tot, "Final with zero", sep = "|")
+heatingResBefore <- paste(tot, "Initial", heating, sep = "|")
+heatingResAfter <- paste(tot, "Final", heating0, sep = "|")
+heatingResAfterZero <- paste(tot, "Final with zero", heating0, sep = "|")
+typeRes <- paste(tot, type, sep = "|")
+locationRes <- paste(tot, location, sep = "|")
+```
+
+#### by building type
+```{r Renovation-Residential-(type)}
+showAreaAndBarPlots(data, vars = typeRes, tot = tot, orderVars = "user", scales = "fixed")
+walk(typeRes, showLinePlots, data = data)
+```
+
+#### by location
+```{r Renovation-Residential-(location)}
+showAreaAndBarPlots(data, vars = locationRes, tot = tot, orderVars = "user", scales = "fixed")
+walk(locationRes, showLinePlots, data = data)
+```
+
+#### by Space heating (initial state)
+```{r Renovation-Residential-(heatingBefore)}
+showAreaAndBarPlots(data, vars = heatingResBefore, tot = totBefore, orderVars = "user", scales = "fixed")
+walk(heatingResBefore, showLinePlots, data = data)
+```
+
+#### by Space heating (final state)
+```{r Renovation-Residential-(heatingAfter)}
+showAreaAndBarPlots(data, vars = heatingResAfter, tot = totAfter, orderVars = "user", scales = "fixed")
+```
+
+#### by Space heating (final state, with zero renovation)
+```{r Renovation-Residential-(heatingAfter)}
+showAreaAndBarPlots(data, vars = heatingResAfterZero, tot = totAfterZero, orderVars = "user", scales = "fixed")
+walk(heatingResAfterZero, showLinePlots, data = data)
```

Comment:
Here the current variable naming means that the total is not shown in the plot, as the variables defined above actually do not exist. The total was still named in this way to preserve the coloring from mip.
Eventually this will require changes in mip to solve the issue:
- either add the variable names of the form "Initial|{hs}" to the mip plotstyle.csv
- or modify mip such that additional tags can be cut from the variable name - or such that the shared part of the variable name is removed (This can potentially change the behavior of mip for many other users!)

Reply:
I found an easy (albeit not perfect) fix for this: If ```tot``` is not specified in the call to ```showAreaAndBarPlots```, then the variable names are shortened by removing their common part, while said common part is considered to be the variable name of the total. However, the values of the total are computed simply as the sum of the plotted variables. So as a result, the area plots display the total and are colored correctly, but the total is not given by the variable ```Renovation|Residential```. but computed as the sum of e.g. ```Renovation|Residential|Initial|{hs}```

Reply:
Then let's go without the specific `tot` and let's see if RSE can look into a mip change.